src/hooks/useUpdateDaoVisibility.ts

```diff
@@ -0,0 +1,119 @@
+import { useCallback, useContext } from 'react'
+import { Contract } from 'ethers'
+import { WORKSPACE_REGISTRY_ADDRESS } from 'src/constants/addresses'
+import { defaultChainId, SupportedChainId } from 'src/constants/chains'
+import useQBContract from 'src/hooks/contracts/useQBContract'
+import { useQuestbookAccount } from 'src/hooks/gasless/useQuestbookAccount'
+import { ApiClientsContext, BiconomyContext, WebwalletContext } from 'src/pages/_app'
+import { bicoDapps, chargeGas, getTransactionDetails, sendGaslessTransaction } from 'src/utils/gaslessUtils'
+
+export default function useUpdateDaoVisibility() {
+	const { nonce } = useQuestbookAccount()
+
+	const { workspace, subgraphClients } = useContext(ApiClientsContext)!
+
+	const workspaceContractGoerli = useQBContract('workspace', SupportedChainId.GOERLI_TESTNET)
+	const workspaceContractPolygon = useQBContract('workspace', SupportedChainId.POLYGON_MAINNET)
+	const workspaceContractOptimism = useQBContract('workspace', SupportedChainId.OPTIMISM_MAINNET)
+	const workspaceContractCelo = useQBContract('workspace', SupportedChainId.CELO_MAINNET)
+
+	const { webwallet, scwAddress } = useContext(WebwalletContext)!
+
+	const { initiateBiconomy, biconomyWalletClients: initialBiconomyWalletClients } = useContext(BiconomyContext)!
+
+	const updateDaoVisibility = useCallback(
+		async(
+			data: { [_: number]: { [_: string]: boolean } },
+			stepsLength: number,
+			setCurrentStep?: (step: number | undefined) => void
+		) => {
+			if(!scwAddress) {
+				return undefined!
+			}
+
+			const chains = Object.keys(data)
+
+			let chainIdx
+			for(chainIdx = 0; chainIdx < chains.length; chainIdx ++) {
+				const chain = chains[chainIdx]
+				const chainId = +chain
+
+				let workspaceContract: Contract
+
+				switch (chainId) {
+				case 5:
+					workspaceContract = workspaceContractGoerli
+					break
+				case 10:
+					workspaceContract = workspaceContractOptimism
+					break
+				case 137:
+					workspaceContract = workspaceContractPolygon
+					break
+				case 42220:
+					workspaceContract = workspaceContractCelo
+					break
+				default:
+					continue
+				}
+
+				setCurrentStep?.(chainIdx * stepsLength)
```

Comment:
instead of using these complex calculations -- have a function called "incrementStep" -- that simply increments & updates currentStep?